docs/pywbemcli/cmdlineinterface.rst

```diff
@@ -529,3 +581,174 @@ Pywbemcli terminates with one of the following program exit codes:
+shell variable be defined that tells the shell to execute tab-completion.
+
+For tab-completion to be activated, the user must register a completion
+function provided by pywbemcli with the shell. Theactivation script is different for
+every shell. The following paragraphs define the activation procedure
+
+Generally activation of pywbemcli involves the following but with different
+formats for each shell type:
+
+* Getting from pywbemcli the body of a completion script for the shell to be
+  activated. This is done by executing a script statement of form
+  ``_PYWBEMCLI_COMPLETE=bash_source pywbemcli`` which causes pywbemcli to
+  return the script it maintains for bash tab-completion.
+* Notifying the shell of this completion script eith an ``eval`` statement or
+  saving the script and notifying the shell later by sourcing the resulting
+  completion script file.
+
+Tab-completion can be activated either by:
+
+* Installing the activation script with an eval statement into the shell
+  startup script.
+* Creating a completion script file and sourcing that file at a later time.
+
+Since much of the logic of activating pywbemcli is based on the shell completion
+logic in the Click package, that package also provides documentation on the
+activation process. Search for "python Click shell completion" in a web browser.
+
+bash - Tab-completion is available with the bash shell for bash versions
+starting with version 4.4 but not all Linux implementations include the
+bash_complete addon so that the user may be required to install the add-on. A
+simple test with a utility like ``ls`` will indicate if tab-completion is
+available  in bash(ex. enter ``ls <TAB>`` to test for the existence of the
+bash_complet addon).
+
+Zsh - TODO
+
+Activation with eval statement
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The following table shows the eval statement for several shells to initialize
+tab completion that would be added to a shell startup file defined in the
+table.
+
+.. table:: Eval statement and proposed startup shell startup file to use for several shells
+   :name: auto-complete-eval-statement
+
+  ======  =======================================  =============================================================
+  Shell   File to insert eval statement            Eval command
+  ======  =======================================  =============================================================
+  bash    ~/.bashrc                                eval "$(_PYWBEMCLI_COMPLETE=bash_source pywbemcli)"
+  zsh     ~/.zshrc                                 eval "$(_PYWBEMCLI_COMPLETE=zsh_source pywbemcli)"
+  fish    ~/.config/fish/completions/foo-bar.fish  eval (env _PYWBEMCLI_COMPLETE=fish_source pywbemcli)
+  ======  =======================================  =============================================================
+
+The above method has some limitations. It may be difficult when virtual
+environments are used and the location of the pywbemcli executable is not known
```

Comment:
For the eval statements and the resulting script outputs, the location of the pywbemcli command does not need to be known. The only requirement is that it is in the path,and that is automatically the case when the pywbemtools packe is installed.

I suggest to remove the sentence about the location difficulty.

Reply:
Changed but still notes that pywbemcli must be in the path

Reply:
Ok